hal/network/ncp_client/sara/sara_ncp_client.cpp

```diff
@@ -1738,6 +1782,23 @@ void SaraNcpClient::checkRegistrationState() {
 int SaraNcpClient::interveneRegistration() {
     CHECK_TRUE(connState_ == NcpConnectionState::CONNECTING, SYSTEM_ERROR_NONE);
 
+    // Check and store if we are using a Twilio SIM based on ICCID
+    if (!iccidChecked_) {
+        char buf[32] = {};
+        auto lenCcid = getIccidImpl(buf, sizeof(buf));
+        CellularNetworkConfig netConfig;
+        CHECK_TRUE(lenCcid > 5, SYSTEM_ERROR_AT_NOT_OK);
+        netConfig = networkConfigForIccid(buf, lenCcid);
+        if (!strcmp(netConfig.apn(), "super")) { // if Twilio Super SIM
+            isTwilioSuperSIM_ = 1;
+        }
+        iccidChecked_ = 1; // cache
+    }
+
+    if (isTwilioSuperSIM_ && millis() - regStartTime_ <= REGISTRATION_TWILIO_HOLDOFF_TIMEOUT) {
+        return 0;
+    }
+
```

Comment:
With this, we will not _start_ running intervene commands before `REGISTRATION_TWILIO_HOLDOFF_TIMEOUT`, but once the time out is expired, we will be running them every 15 sec. Do we want to run the intervention commands every 15 sec or `REGISTRATION_INTERVENTION_TIMEOUT` sec?

Reply:
Talked with Andrey. We stick to this. thanks